### DISCUSSION TITLE
navigation.min.js loading priority

### DISCUSSION THREAD

**Discussion by luukee** [wprig/wprig#834] [General]
Curious if there's any way to add a `prefetch="high"` priority to the `navigation.min.js`?

Thank you!
Luke

**Comment by robruiz** [comment]
To my knowledge there is no prefetch attribute. There is only the rel
attribute which accepts prefetch and other values. I don't think there is a
web spec for prioritizing prefetches. If you want something to load before
something else, you should look into the priority argument in
wp-enqueue-scripts. For the record I think navigation is already prefetches
with rel. In the scripts component you can assign prefetch context in the
scripts array.

**Comment by robruiz** [comment]
Where you actually referring to this?
https://web.dev/fetch-priority/

**Comment by luukee** [comment]
Thanks @robruiz. Yes, https://web.dev/fetch-priority/ is where I found the `prefetch` info. 

I was inspecting my page loading sequence in Chrome devtools Performance insights and curious if I could load any above-the-fold elements quicker. I was also reviewing the loading timeline in [LogRocket](https://logrocket.com/) and saw the navigation CLS was pretty high (not in the below screenshot).

![Screenshot 2023-05-19 at 7 22 12 AM](https://github.com/wprig/wprig/assets/30611098/32c5a221-884e-45cc-8868-9154e0fc8227)

Anyways, I thought it might be `wp_script_add_data` from `inc/Accessibility/Component.php` here:
```
	public function action_enqueue_navigation_script() {

		// If the AMP plugin is active, return early.
		if ( wp_rig()->is_amp() ) {
			return;
		}

		// Enqueue the navigation script.
		wp_enqueue_script(
			'wp-rig-navigation',
			get_theme_file_uri( '/assets/js/navigation.min.js' ),
			array(),
			wp_rig()->get_asset_version( get_theme_file_path( '/assets/js/navigation.min.js' ) ),
			false
		);
		wp_script_add_data( 'wp-rig-navigation', 'async', true );
		wp_script_add_data( 'wp-rig-navigation', 'precache', true );
		wp_localize_script(
			'wp-rig-navigation',
			'wpRigScreenReaderText',
			array(
				'expand'   => __( 'Expand child menu', 'wp-rig' ),
				'collapse' => __( 'Collapse child menu', 'wp-rig' ),
			)
		);
	}
```
. . . but I see `async` & `precache` are already there, so it's probably no big deal. Appreciate you getting into this with me :)
